src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+        String description = String.join(" ",
+                Arrays.copyOfRange(inputSplit, 1, inputSplit.length));
+
+        if (description.isEmpty()) {
+            throw new DukeException("Please input the description!");
+        }
+
+        Todo newTodo = new Todo(description);
+        taskManager.addTask(newTodo);
+    }
+
+    private static void handleDeadline(String[] inputSplit) throws DukeException {
+        int indexOfBy = Arrays.asList(inputSplit).indexOf("/by");
+
+        if (indexOfBy == -1) {
+            throw new DukeException("Please input the description and/or deadline of the task!");
+        }
+
+        String description = String.join(" ", Arrays.copyOfRange(inputSplit, 1, indexOfBy));
+        String by = String.join(" ",
+                Arrays.copyOfRange(inputSplit, indexOfBy + 1, inputSplit.length));
+
+        if (description.isEmpty() && by.isEmpty()) {
+            throw new DukeException("Please input the description and deadline of the task!");
+        } else if (description.isEmpty()) {
+            throw new DukeException("Please input the description of the task!");
+        } else if (by.isEmpty()) {
+            throw new DukeException("Please input the deadline of the task!");
+        }
+
+        Deadline newDeadline = new Deadline(description, by);
+        taskManager.addTask(newDeadline);
+    }
+
+    private static void handleEvent(String[] inputSplit) throws DukeException {
+        int indexOfAt = Arrays.asList(inputSplit).indexOf("/at");
+
+        if (indexOfAt == -1) {
+            throw new DukeException("Please input the description and/or duration of the event!");
+        }
+
+        String description = String.join(" ", Arrays.copyOfRange(inputSplit, 1, indexOfAt));
+        String at = String.join(" ",
+                Arrays.copyOfRange(inputSplit, indexOfAt + 1, inputSplit.length));
+
+        if (description.isEmpty() && at.isEmpty()) {
+            throw new DukeException("Please input the description and duration of the event!");
+        } else if (description.isEmpty()) {
+            throw new DukeException("Please input the description of the event!");
+        } else if (at.isEmpty()) {
+            throw new DukeException("Please input the duration of the event!");
+        }
+
+        Event newEvent = new Event(description, at);
+        taskManager.addTask(newEvent);
+    }
+
+    private static void handleDelete(String[] inputSplit) throws DukeException {
+        try {
+            int indexOfTask = Integer.parseInt(inputSplit[1]) - 1;
+            taskManager.deleteTask(indexOfTask);
+        } catch (NumberFormatException e) {
+            throw new DukeException("Please input a valid index of task!");
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please input the index of task you want to delete!");
+        }
+    }
+
     public static void main(String[] args) {
```

Comment:
The method main() seems to be more than 30 lines of code, maybe could consider extracting the while true loop to another method.